### PR TITLE
us-newsletter-hide-soft-opt-in

### LINKED ISSUE #15877: Remove marketing soft opt in for US: Hide toggle and apply removed soft opt in (client-side logic)
## Overview
Implement the UI logic for removing the marketing soft opt in for US newsletter sign-ups. When enabled for signed out US users, the marketing toggle is hidden and the user is automatically opted into `similar_guardian_products`.

## Acceptance Criteria
- [x] Add a helper to check whether the **US soft opt-in switch** is enabled (via `window.guardian.config.switches.us-signup-hide-marketing-toggle`)
- [x] Use the `useCountryCode` hook to determine country
- [x] In all sign-up surfaces (`SecureSignup`, `NewsletterSignupForm`, `ManyNewsletterSignUp`):
	- If the switch is OFF, current behaviour (toggle visible for all users)
	- If the switch is ON and `countryCode === 'US'`, hide the toggle and always POST `marketing=true`
	- All other countries: toggle visible as normal
	- No changes for signed-in users
- [x] Treat `countryCode === undefined` (pending) as non-US — show toggle while resolving
- [x] Add/update unit tests to cover US, non-US, and pending states
- [x] Ensure that a `marketingOptInHidden` field is included in the sign-up payload, alongside `countryCode: 'US'`, when the hidden soft opt-in is applied
- [x] Analytics tracking: the `marketingOptInType` field sent in Ophan tracking events should reflect the soft opt-in case
  - When the toggle is hidden (US soft opt-in applied), send `marketingOptInType: 'similar-guardian-products-optin-hidden-us'
  - When the toggle is visible and the user opts in, send `similar-guardian-products-optin`
  - When the toggle is visible and the user opts out, send `similar-guardian-products-optout`
  - For signed-in users, omit `marketingOptInType` from the tracking payload
  - Applies to all three surfaces: `SecureSignup`, `NewsletterSignupForm`, `ManyNewsletterSignUp`

## Additional Notes
- The switch for this functionality is currently referred to as `us-signup-hide-marketing-toggle` in the codebase; ensure consistent usage across code, tests, and stories.
- Parent issue: [guardian/email-rendering#930](https://github.com/guardian/email-rendering/issues/930)

### PR DESCRIPTION
**Pull request by georgerichmond**
# US newsletter marketing soft opt-in

## What does this change?

Implements the UI logic for hiding the marketing consent toggle for signed-out US users when the `usSignupHideMarketingToggle` switch is enabled.

- Adds a `useHideMarketingToggleForCountry` hook that checks the switch and the user's country code via `useCountryCode`
- When the switch is **on** and `countryCode === 'US'` and the user is **signed out**, the marketing toggle is hidden and `marketing=true` is always included in the sign-up payload
- When the switch is **off**, or the user is signed in, or the country is anything other than `US` (including `undefined` while resolving), existing behaviour is unchanged
- A `marketingOptInHidden: true` field is appended to the sign-up request payload in the hidden case so the backend can identify implicit opt-ins
- `marketingOptInType` is included in Ophan tracking events on all three surfaces:
    - `similar-guardian-products-optin-hidden-us` — toggle hidden (US soft opt-in)
    - `similar-guardian-products-optin` — toggle visible, user opted in
    - `similar-guardian-products-optout` — toggle visible, user opted out
    - Omitted entirely for signed-in users
- Applies consistently across all three sign-up surfaces: `SecureSignup`, `NewsletterSignupForm`, and `ManyNewsletterSignUp`
- Mocks `useCountryCode` in Storybook preview and adds a `USHideMarketingToggle` story to `NewsletterSignupForm`

## Why?

US consumer protection regulations mean we should not present a marketing opt-in toggle to US newsletter sign-ups. Instead, users are silently enrolled in `similar_guardian_products` — i.e. a soft opt-in. This is gated behind a switch so it can be enabled/disabled independently of a deploy.

Parent issue: [guardian/email-rendering#930](https://github.com/guardian/email-rendering/issues/930)

## Screenshots

_No visible UI change for non-US users. For US signed-out users, the marketing toggle is hidden — no before/after screenshot is provided as this is switch-gated._

| Before | After |
| ------ | ----- |
| Marketing toggle visible for all signed-out users | Marketing toggle hidden for signed-out US users (switch on) |

### CODE REVIEW

File: dotcom-rendering/src/components/ManyNewsletterSignUp.island.tsx [hunk 1]
```diff
@@ -222,7 +231,7 @@ export const ManyNewsletterSignUp = ({
 	}, [status]);
 
 	useEffect(() => {
-		if (isSignedIn !== 'Pending' && !isSignedIn) {
+		if (isSignedIn !== true) {
```
[the remark this answers is not in the document]
Comment (georgerichmond) [reply dotcom-rendering/src/components/ManyNewsletterSignUp.island.tsx]: No we want to show the box by default until it's removed

File: dotcom-rendering/src/components/SecureSignup.island.tsx [hunk 2]
```diff
@@ -251,6 +270,9 @@ export const SecureSignup = ({
 	);
 	const isSignedIn = useIsSignedIn();
 	const authStatus = useAuthStatus();
+	const hideMarketingToggle = useHideMarketingToggleForCountry();
+	const marketingOptInHiddenForCountry =
+		hideMarketingToggle && isSignedIn !== true;
```
[the remark this answers is not in the document]
Comment (georgerichmond) [reply dotcom-rendering/src/components/SecureSignup.island.tsx]: edge case - is more important to show until resolved

File: dotcom-rendering/src/lib/useNewsletterSignupForm.ts [hunk 3]
```diff
@@ -270,19 +298,35 @@ export const useNewsletterSignupForm = (
...
+			const effectiveMarketingOptIn = getEffectiveMarketingOptIn({
+				marketingOptInHiddenForCountry,
+				isSignedIn: isSignedInRef.current,
+				marketingOptIn: marketingOptInRef.current,
+			});
+			const marketingOptInType = getMarketingOptInType({
+				marketingOptInHiddenForCountry,
+				isSignedIn: isSignedInRef.current,
+				effectiveMarketingOptIn,
+			});
```
[the remark this answers is not in the document]
Comment (georgerichmond) [reply dotcom-rendering/src/lib/useNewsletterSignupForm.ts:330]: Edge case - is more important to show the box until resolved

File: dotcom-rendering/src/components/SecureSignup.island.test.tsx [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+			).toBeInTheDocument(),
+		);
+		await testUser.type(
+			screen.getByLabelText('Enter your email address'),
+			'reader@example.com',
+		);
+		await testUser.click(screen.getByRole('button', { name: 'Sign up' }));
+	};
+
+	describe('US hide marketing toggle (usSignupHideMarketingToggle switch)', () => {
```
Comment (rBangay) [inline dotcom-rendering/src/components/SecureSignup.island.test.tsx]: Hey 👋  ... do we need this describe block, there is already a parent describe and this describe has no siblings

File: dotcom-rendering/src/components/SecureSignup.island.test.tsx [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+		recaptchaBehaviour = (_handle, props) =>
+			props.onChange?.('test-recaptcha-token');
+
+		(useIsSignedIn as jest.Mock).mockReturnValue(false);
+		(useAuthStatus as jest.Mock).mockReturnValue({ kind: 'SignedOut' });
+		(useBrowserId as jest.Mock).mockReturnValue('test-browser-id');
+		(useCountryCode as jest.Mock).mockReturnValue(undefined);
+
+		pageConfig.ajaxUrl = 'https://api.nextgen.guardianapps.co.uk';
+		pageConfig.idApiUrl = 'https://idapi.nextgen.guardianapps.co.uk';
```
Comment (rBangay) [inline dotcom-rendering/src/components/SecureSignup.island.test.tsx]: Do we need these defined here if just below we are mocking fetch responses?

File: dotcom-rendering/src/components/SecureSignup.island.tsx [hunk 6]
```diff
@@ -209,13 +224,17 @@ const sendTracking = (
 	eventDescription: NewsletterEventDescription,
 	renderingTarget: RenderingTarget,
 	abTest?: AbTest,
+	extraDetails?: Record<string, unknown>,
```
Comment (rBangay) [inline dotcom-rendering/src/components/SecureSignup.island.tsx:228]: Hey, both `abTest` and `extraDetails` are optional params. If there were a scenario where you want to passthrough extraDetails but not abTest. you would have to pass through a null value for abTest. Perhaps the optional params could be encapsulated in an object (optional) then you could pass one but not the other.

File: dotcom-rendering/src/components/SecureSignup.island.tsx [hunk 7]
```diff
@@ -274,15 +301,32 @@ export const SecureSignup = ({
...
+		);
 
 		const formData = buildFormData(
 			emailAddress,
 			newsletterId,
 			token,
-			marketingOptIn,
+			effectiveMarketingOptIn,
 			browserId,
+			marketingOptInHiddenForCountry ? true : undefined,
```
Comment (rBangay) [inline dotcom-rendering/src/components/SecureSignup.island.tsx:332]: if eslint doesn't complain, this could probably be

```
marketingOptInHiddenForCountry || undefined
```

File: dotcom-rendering/src/lib/newsletter-marketing-opt-in.ts [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+export const getEffectiveMarketingOptIn = ({
+	marketingOptInHiddenForCountry,
+	isSignedIn,
+	marketingOptIn,
+}: {
+	marketingOptInHiddenForCountry: boolean;
+	isSignedIn: boolean | 'Pending';
+	marketingOptIn: boolean | undefined;
+}): boolean | undefined => {
+	// Only apply the US soft opt-in when we know the user is signed out
```
Comment (rBangay) [inline dotcom-rendering/src/lib/newsletter-marketing-opt-in.ts]: Pedant comment: `marketingOptInHiddenForCountry` is country agnostic, but this comment explicitly mentions the US

File: dotcom-rendering/src/lib/useNewsletterSignupForm.ts [hunk 9]
```diff
@@ -26,6 +30,7 @@ const buildFormData = (
 	token: string,
 	marketingOptIn?: boolean,
 	browserId?: string,
+	marketingOptInHiddenForCountry?: boolean,
```
Comment (rBangay) [inline dotcom-rendering/src/lib/useNewsletterSignupForm.ts]: possibly a non issue, but you will have to pass through null values for marketingOptIn and browserId if you want to pass through a value for marketingOptInHiddenCountry but not the other 1 or 2 params

**Review by rBangay — APPROVED** [review]
